force-app/main/default/classes/ADDR_Addresses_UTIL_TEST.cls

```diff
@@ -0,0 +1,1341 @@
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentAddressField() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+        contact.MailingCity = 'Something else';
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameFalseDifferentStateCode() {
+        /*Almost no way we can test state code/country code here
+         This need to be refactored so that we can override the returned result*/
+        //TODO(Refactor): Make isStateCountryPicklistsEnabled overridable
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNotNullContactNullAccount() {
+        Contact contact = UTIL_UnitTestData_TEST.getContact();
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(contact, null);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNullContactNotNullAccount() {
+        List<Account> accounts = UTIL_UnitTestData_TEST.getMultipleTestAccounts(
+            1,
+            UTIL_Describe_API.getBizAccRecTypeID()
+        );
+
+        Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(null, accounts[0]);
+        Test.stopTest();
+
+        System.assertEquals(false, isContactAccAddrInfoSame);
+    }
+
+    @isTest
+    static void isContactAccAddrInfoSameNullContactNullAccount() {
+        //We don't have guard clause for the scenario when both contact and account are null
+        //Thus this will hit null pointer exception. Given the static method, we should refactor this.
+        //TODO(Refactor): add guard clause for contact == null $$ account == null
+
+        /*Test.startTest();
+        Boolean isContactAccAddrInfoSame = ADDR_Addresses_UTIL.isContactAccAddrInfoSame(null, null);
+        Test.stopTest();
+
+        System.assertEquals(true, isContactAccAddrInfoSame);*/
+    }
+
+    /**************************************************************************************************************************
+     ****************************************************** isAddressChanged **************************************************
+     **************************************************************************************************************************/
+    @isTest
```

Comment:
This is one of those use cases where technically we should be validating every potential scenario for address changing. This is why I tend to make a test and assertion helper for those to make the methods really small. We don't have to go that far this time, but it's worth noting.

Given that we're using case sensitivity, it might be worth changing the case for the last letter of the address to really capture that kind of change.